firebase-perf/src/test/java/com/google/firebase/perf/application/AppStateMonitorTest.java

```diff
@@ -722,6 +723,35 @@ public void activityStateChanges_multipleClients_callbackCalledOnEachClient() {
     assertThat(clientState.get(client3)).isEqualTo(ApplicationProcessState.BACKGROUND);
   }
 
+  @Test
```

Comment:
Can we add test cases for the PubSub setup? Eg. adding subscriber multiple times, removing multiple times to make sure we do not crash?

Reply:
Added two more test case:
 - appColdStart_multipleSubscriber_callbackIsCalled
 - appColdStart_singleSubscriberRegistersForMultipleTimes_oneCallbackIsCalled